webapp/views/App/views/Data/Charts/state/chartTypes/pie/index.js

```diff
@@ -29,88 +19,164 @@ const pie = {
         subtitle: 'Here we config the query of the pie',
         type: 'container',
         blocks: {
-          groupBy: {
-            id: 'groupBy',
-            title: 'Group by',
-            subtitle: 'Select the dimension to group the data',
-            type: 'input',
-            onUpdateSpec:
-              ({ spec, onUpdateSpec }) =>
-              ({ dimension }) => {
-                if (!dimension) {
-                  const newSpec = {
-                    ...spec,
-                  }
-                  delete newSpec.encoding.color
-                  onUpdateSpec(A.stringify(newSpec))
-                  return
-                }
+          groupBy: GroupByBlock({
+            valuesToSpec: ({ value = [], spec = {} }) => {
+              const transform = {
+                calculate: `${value.map((val) => `datum.${val.value}`).join("+','+")}`,
```

Comment:
the way to determine the "calculate" is repeated several times. Can you please extract it into some helper function?